### PR TITLE
memory_opt_check_align: require aligned memory addresses in State

### PR DESCRIPTION
**Pull request by naure**
- Memory alignment is checked by a lookup in the State circuit.
- LOG addresses start at 0 so we don’t realignment in the Copy circuit (front_mask=0).

### CODE REVIEW

File: zkevm-circuits/src/copy_circuit.rs [hunk 1]
```diff
@@ -418,20 +418,22 @@ impl<F: Field> SubCircuitConfig<F> for CopyCircuitConfig<F> {
...
-                cb.require_zero(
-                    "back_mask=0 at the start of the next word",
-                    and::expr([
-                        is_continue.expr(),
-                        back_mask_next.expr(),
-                    ]),
-                );*/
             });
 
+            /* Note: other words may be completely masked, because reader and writer may have different word counts. A fully masked word is a no-op, not contributing to value_acc, and its word_rlc equals word_rlc_prev.
```
Comment (DreamWuGit) [inline zkevm-circuits/src/copy_circuit.rs:428]: I think we don't have full mask word, if copy len is zero. no any mask.

**Comment by DreamWuGit** [comment]
some state test fails, it should be caused by old addr not divide by 32,  update addr should pass.
    state_circuit::test::bad_initial_memory_value
    state_circuit::test::nonlexicographic_order_tag
    state_circuit::test::read_inconsistency
    state_circuit::test::state_circuit_simple_2

File: zkevm-circuits/src/copy_circuit.rs (hunk 1 above)
Comment (naure) [reply zkevm-circuits/src/copy_circuit.rs:428]: Here is an example:
- Source range [31..33], requires two words.
- Destination range [0..2], one word + one fully-masked word.

Comment (DreamWuGit) [reply zkevm-circuits/src/copy_circuit.rs:428]: I misunderstood all words are masked.

**Comment by naure** [comment]
Fixed. Plus new negative tests.

**Pull request merged**